sdk/keyvault/keyvault/inc/az_keyvault_create_key_options.h

```diff
@@ -8,16 +8,106 @@
 
 #include <_az_cfg_prefix.h>
 
+static uint8_t const AZ_KEYVAULT_MAX_KEY_OPERATIONS = 6;
+
+AZ_NODISCARD AZ_INLINE az_span az_keyvault_key_operation_decrypt() { return AZ_STR("decrypt"); }
+AZ_NODISCARD AZ_INLINE az_span az_keyvault_key_operation_encrypt() { return AZ_STR("encrypt"); }
+AZ_NODISCARD AZ_INLINE az_span az_keyvault_key_operation_sign() { return AZ_STR("sign"); }
+AZ_NODISCARD AZ_INLINE az_span az_keyvault_key_operation_unwrapKey() { return AZ_STR("unwrapKey"); }
+AZ_NODISCARD AZ_INLINE az_span az_keyvault_key_operation_verify() { return AZ_STR("verify"); }
+AZ_NODISCARD AZ_INLINE az_span az_keyvault_key_operation_wrapKey() { return AZ_STR("wrapKey"); }
+
+typedef struct {
+  az_span operations[6];
+  uint8_t size;
+} az_keyvault_create_key_options_key_operation;
+
 typedef struct {
   az_optional_bool enabled;
+  az_keyvault_create_key_options_key_operation key_operations;
   /* TODO: adding next options
   Datetime not_before;
   Datetime expires_on
   List tags
-  List keyOperations
   */
 } az_keyvault_create_key_options;
 
+/**
+ * @brief check if there is at least one operation
+ *
+ */
+AZ_NODISCARD AZ_INLINE bool az_keyvault_create_key_options_is_empty(
+    az_keyvault_create_key_options const * self) {
+  return self->key_operations.size == 0;
+}
+
+/**
+ * @brief check if operation array is full
+ *
+ */
+AZ_NODISCARD AZ_INLINE bool az_keyvault_create_key_options_is_full(
+    az_keyvault_create_key_options * const self) {
+  return self->key_operations.size >= AZ_KEYVAULT_MAX_KEY_OPERATIONS;
+}
+
+/**
+ * @brief set size of operations to 0 so it can be written again
+ *
+ */
+AZ_NODISCARD AZ_INLINE az_result
+az_keyvault_create_key_options_clear(az_keyvault_create_key_options * const self) {
```

Comment:
Clearing should set the key_operations field to (az_keyvault_create_key_options_key_operation) {0}.  That way the spans array and the size get reset.